src/app/fdctl/run/tiles/fd_repair.c

```diff
@@ -0,0 +1,675 @@
+  eth_ip_udp_t * hdr = (eth_ip_udp_t *)ctx->repair_buffer;
+
+  fd_repair_peer_addr_t peer_addr;
+  peer_addr.l = 0;
+  peer_addr.addr = ip;
+  peer_addr.port = fd_ushort_bswap( hdr->udp->net_sport );
+
+  fd_repair_settime( ctx->repair, fd_log_wallclock() );
+  fd_repair_continue( ctx->repair );
+  fd_repair_recv_packet( ctx->repair, ctx->repair_buffer + hdr_sz, ctx->repair_buffer_sz - hdr_sz, &peer_addr );
+}
+
+static void
+privileged_init( fd_topo_t *      topo    FD_PARAM_UNUSED,
+                 fd_topo_tile_t * tile    FD_PARAM_UNUSED,
+                 void *           scratch FD_PARAM_UNUSED) {
+
+}
+
+static void
+during_housekeeping( void * _ctx ) {
+  fd_repair_tile_ctx_t * ctx = (fd_repair_tile_ctx_t *)_ctx;
+
+  fd_mcache_seq_update( ctx->store_out_sync, ctx->store_out_seq );
+
+  fd_repair_settime( ctx->repair, fd_log_wallclock() );
+  fd_repair_continue( ctx->repair );
+}
+
+static void
+unprivileged_init( fd_topo_t *      topo,
+                   fd_topo_tile_t * tile,
+                   void *           scratch ) {
+  fd_flamenco_boot( NULL, NULL );
+
+  if( FD_UNLIKELY( tile->in_cnt != 4 ||
+                   strcmp( topo->links[ tile->in_link_id[ NET_IN_IDX     ] ].name, "net_repair")     ||
+                   strcmp( topo->links[ tile->in_link_id[ CONTACT_IN_IDX ] ].name, "gossip_repair" ) ||
+                   strcmp( topo->links[ tile->in_link_id[ STAKE_IN_IDX ] ].name,   "stake_out" )     ||
+                   strcmp( topo->links[ tile->in_link_id[ STORE_IN_IDX ] ].name,   "store_repair" ) ) ) {
+    FD_LOG_ERR(( "Repair tile has none or unexpected input links %lu %s %s",
+                 tile->in_cnt, topo->links[ tile->in_link_id[ 0 ] ].name, topo->links[ tile->in_link_id[ 1 ] ].name ));
+  }
+
+  if( FD_UNLIKELY( tile->out_cnt != 1 ||
+                   strcmp( topo->links[ tile->out_link_id[ STORE_OUT_IDX ] ].name, "repair_store" ) ) ) {
+    FD_LOG_ERR(( "repair tile has none or unexpected output links %lu %s %s",
+                 tile->out_cnt, topo->links[ tile->out_link_id[ 0 ] ].name, topo->links[ tile->out_link_id[ 1 ] ].name ));
+  }
+
+  if( FD_UNLIKELY( tile->out_link_id_primary == ULONG_MAX ) )
+    FD_LOG_ERR(( "repair tile has no primary output link" ));
+
+  /* Scratch mem setup */
+
+  FD_SCRATCH_ALLOC_INIT( l, scratch );
+  fd_repair_tile_ctx_t * ctx = FD_SCRATCH_ALLOC_APPEND( l, alignof(fd_repair_tile_ctx_t), sizeof(fd_repair_tile_ctx_t) );
+  ctx->repair = FD_SCRATCH_ALLOC_APPEND( l, fd_repair_align(), fd_repair_footprint() );
+
+  void * smem = FD_SCRATCH_ALLOC_APPEND( l, fd_scratch_smem_align(), fd_scratch_smem_footprint( SMAX ) );
+  void * fmem = FD_SCRATCH_ALLOC_APPEND( l, fd_scratch_fmem_align(), fd_scratch_fmem_footprint( SDEPTH ) );
+
+  FD_TEST( ( !!smem ) & ( !!fmem ) );
+  fd_scratch_attach( smem, fmem, SMAX, SDEPTH );
+  
+  ulong scratch_top = FD_SCRATCH_ALLOC_FINI( l, 1UL );
+  if( FD_UNLIKELY( scratch_top > (ulong)scratch + scratch_footprint( tile ) ) )
```

Comment:
Prefer to do this check at the bottom of the init function like the other tiles (in case someone adds another scratch alloc below)